Sources/MapboxCoreNavigation/RouteController.swift

```diff
@@ -273,6 +273,42 @@ open class RouteController: NSObject {
         }
         // Check for faster route proactively (if reroutesProactively is enabled)
         refreshAndCheckForFasterRoute(from: location, routeProgress: routeProgress)
+        
+        verifyNavigationTilesVersion()
+    }
+    
+    private var wantsSwitchTilesVersion = false
+    private var wantsRestoreTilesVersion = false
+    
+    private func verifyNavigationTilesVersion() {
```

Comment:
Looks like a copy-paste. There should be a way to generalize this code

Reply:
Yeah, I tried, but since `RouteController` and `PassiveLocationDataSource` have different delegates to call and different completion logic to handle, generalisation looked way uglier than code duplication.

Reply:
Thought about this again... It holds some logic(like selecting the latest version) and I would recommend reconsidering moving this to a class. Maybe both delegates (`PassiveLocationDataSourceDelegate` and `RouterDelegate`) can inherit `NavigationTilesVersioningDelegate`?